test/e2e/generate_kube_test.go

```diff
@@ -849,6 +849,62 @@ var _ = Describe("Podman kube generate", func() {
 		Expect(inspect.OutputToString()).To(ContainSubstring(vol1))
 	})
 
+	It("with subpath volume", func() {
+		var args []string
+		volName := "vol-test1"
+		mkvol := podmanTest.Podman([]string{"volume", "create", volName})
+		mkvol.WaitWithDefaultTimeout()
+		Expect(mkvol).Should(ExitCleanly())
+
+		mountPoint := "/mnt"
+		filepath1 := "etc/etcfile"
+		dir1 := filepath.Join(mountPoint, "etc")
+		inFile1 := filepath.Join(mountPoint, filepath1)
+		exec1 := fmt.Sprintf("mkdir -p %s; touch %s", dir1, inFile1)
+
+		filepath2 := "var/varfile"
+		dir2 := filepath.Join(mountPoint, "var")
+		inFile2 := filepath.Join(dir2, "varfile")
+		exec2 := fmt.Sprintf("mkdir -p %s; touch %s", dir2, inFile2)
+
+		popvol := podmanTest.Podman([]string{"run", "-v", volName + ":" + mountPoint,
+			ALPINE, "sh", "-c", exec1 + ";" + exec2,
+		})
+		popvol.WaitWithDefaultTimeout()
+		Expect(popvol).Should(ExitCleanly())
+
+		mountTemplate := "type=volume,source=%s,volume-subpath=/%s,target=/%s"
+		args = []string{"run", "-d", "--pod", "new:test1", "--name", "test-ctr",
+			"--mount", fmt.Sprintf(mountTemplate, volName, filepath1, filepath1),
+			"--mount", fmt.Sprintf(mountTemplate, volName, filepath2, filepath2),
+			CITEST_IMAGE, "top"}
+
+		session1 := podmanTest.Podman(args)
+		session1.WaitWithDefaultTimeout()
+		Expect(session1).Should(ExitCleanly())
+
+		outputFile := filepath.Join(podmanTest.RunRoot, "pod.yaml")
+		kube := podmanTest.Podman([]string{"kube", "generate", "test1", "-f", outputFile})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(ExitCleanly())
+
+		b, err := os.ReadFile(outputFile)
```

Comment:
this could be avoided if you removed the `-f` above and then examine the output?

Reply:
Indeed it does. I had to use the `kube.OutputToStringArray()` and join them with `\n` but combined with `podmanTest.PodmanExitCleanly()` some code could be removed. Thanks!